### ISSUE TITLE
Add option to pass config object instead of filepath to Launcher

### ISSUE THREAD

**Issue by vbyjsue** [webdriverio/webdriverio#1914]
## The problem

We use our gui-test framework in a fairly big organisation and try to make running it as simple as possible. It is started by passing arguments to a file which then sets up config accordingly by passing options to Launcher:
`let wdio = new Launcher(opts.configFile, opts)`

This makes it hard to follow which configurations are set in config file or passed in as options. It is also not possible to pass in cababilities.

With the possiblity to pass a config object instead of file, the default config could be kept in file then read and altered depending on arguments and passed to Launcher.

## Environment

* WebdriverIO version: 4.6.2
* Node.js version: 6.9.2
* Standalone mode or wdio testrunner: testrunner
* if wdio testrunner, running synchronous or asynchronous tests: sync

## Details

This is what I would like to do:
1. Start-file that accepts arguments. `node run start-file.js [test-env] [formfactor] [debug] ....`

start-file.js:
2. Read config-file
3. Do complex config altering depending on args.
4. Pass config object to Launcher.

**Comment by christian-bromann** [comment]
Instead of passing the config object to the launcher just create a temporary config file using the [tmp](https://www.npmjs.com/package/tmp) module and pass the path to the launcher.

**Comment by vbyjsue** [comment]
The intention was to avoid having to create a new file, but not a major problem . Thanks for fast feedback.

**Comment by christian-bromann** [comment]
> The intention was to avoid having to create a new file

@vbyjsue wdio needs a file to read its options from because it can't pass all data types (e.g. functions) to the runner child process. It is easier to just create a tmp file than hacking something together that would support that.

**Comment by Chris-Eckersley** [comment]
@christian-bromann, @vbyjsue, my co-worker found a solution by passing in the path to a dummy file with an empty object. Then pass in the dummy path and your object as the `opts` arg for the launcher. So far it's working.

Example: 

`Const wdioConfigObject = {...};
const CONFIGPATH = './config/dummy.js';
const wdio = new Launcher(CONFIGPATH, wdioConfigObject);`

**Comment by bschlenk** [comment]
This prevents us from reusing existing modules written in typescript. We can run our test launcher with `ts-node`, but it doesn't seem to transpile files that are required from within node_modules. We get the error `Cannot use import statement outside a module`.
